GrpcClient/ShiftGrpcRepository.cs

```diff
@@ -0,0 +1,257 @@
+            using var channel = GrpcChannel.ForAddress(_grpcAddress);
+            var client = new Shift.ShiftClient(channel);
+            var request = new Id { Id_ = id };
+            var reply = await client.GetSingleShiftByIdAsync(request);
+            return grpcShiftObject(reply);
+        }
+        catch (RpcException e)
+        {
+            if (e.StatusCode == StatusCode.NotFound)
+            {
+                throw new ArgumentException(e.Message);
+            }
+
+            throw;
+        }
+    }
+
+    public async Task<bool> IsShiftInRepository(long id)
+    {
+        try
+        {
+            using var channel = GrpcChannel.ForAddress(_grpcAddress);
+            var client = new Shift.ShiftClient(channel);
+            var reply = await client.IsShiftInRepositoryAsync(new Id { Id_ = id });
+            return reply.Boolean_;
+        }
+        catch (RpcException e)
+        {
+            if (e.StatusCode == StatusCode.NotFound)
+            {
+                throw new ArgumentException(e.Message);
+            }
+
+            throw;
+        }
+    }
+
+    public async Task<Entities.Shift> AssignEmployeeToShift(long shiftId, long employeeId)
+    {
+        try
+        {
+            using var channel = GrpcChannel.ForAddress(_grpcAddress);
+            var client = new Shift.ShiftClient(channel);
+            var assignEmployeeRequest = new ShiftEmployeePair
+            {
+                ShiftId = shiftId,
+                EmployeeId = employeeId
+            };
+
+            Console.WriteLine($"Assigning Employee ID: {employeeId} to Shift ID: {shiftId}");
+            var reply = await client.AssignEmployeeToShiftAsync(assignEmployeeRequest);
+            Console.WriteLine("Assignment successful, fetching updated shift");
+
+            var updatedShift = await client.GetSingleShiftByIdAsync(new Id { Id_ = shiftId });
+            Console.WriteLine($"Updated Shift: {updatedShift.Id}, Employee ID: {employeeId}");
+
+            return grpcShiftObject(updatedShift);
+        }
+        catch (RpcException e)
+        {
+            Console.WriteLine($"Error: {e.StatusCode}, Message: {e.Message}");
+            if (e.StatusCode == StatusCode.NotFound)
+            {
+                throw new ArgumentException(e.Message);
+            }
+
+            throw;
+        }
+    }
+
+    public async  Task<Entities.Shift> UnassignEmployeeToShift(long shiftId, long employeeId)
+    {
+        try
+        {
+            var unassignEmployeeRequest = new ShiftEmployeePair
+            {
+                ShiftId = shiftId,
+                EmployeeId = employeeId
+            };
+            
+            using var channel = GrpcChannel.ForAddress(_grpcAddress);
+            var client = new Shift.ShiftClient(channel);
+            var unassignRequest = new ShiftEmployeePair { ShiftId = shiftId, EmployeeId = employeeId};
+            var reply = await client.UnAssignEmployeeFromShiftAsync(unassignEmployeeRequest);
```

Comment:
Is it a typo or on purpose? UnAssignEmployeeFromShiftAsync/UnassignEmployeeToShift

Reply:
O, just noticed one is To and other is From. Confusing stuff